snapcraft/parts/lifecycle.py

```diff
@@ -129,24 +138,29 @@ def run(command_name: str, parsed_args: "argparse.Namespace") -> None:
     emit.trace(f"command: {command_name}, arguments: {parsed_args}")
 
     snap_project = get_snap_project()
-    yaml_data = process_yaml(snap_project.project_file)
-    parse_info = _extract_parse_info(yaml_data)
+    yaml_data = load_yaml(snap_project.project_file)
+    build_plan = get_build_plan(yaml_data)
 
-    if parsed_args.provider:
-        raise errors.SnapcraftError("Option --provider is not supported.")
+    for build_on, build_to in build_plan:
+        emit.progress(f"running with build_on: {build_on} and build_to: {build_to}")
+        yaml_data_for_arch = process_yaml(yaml_data, build_to)
+        parse_info = _extract_parse_info(yaml_data_for_arch)
 
-    # Register our own plugins
-    plugins.register()
+        if parsed_args.provider:
```

Comment:
this should be right below the first emit trace...

Reply:
Moved out of the loop, but it has to go after `yaml_load`, since that function raises `LegacyFallback`.